File: scripts/check.js
const fs = require('fs');

try {
  let buf = fs.readFileSync('parsed.json');
  let text = buf.toString('utf16le');
  
  // Find "tagStats"
  const start = text.indexOf('"tagStats":[');
  if (start === -1) {
    console.log("No tagStats found");
    process.exit(0);
  }
  
  const end = text.indexOf(']', start);
  const arrStr = text.substring(start + '"tagStats":'.length, end + 1);
  
  // Clean it up
  const cleanStr = arrStr.replace(/[\u0000-\u001F]/g, '');
  const tags = JSON.parse(cleanStr);
  
  const review = tags.filter(t => t.fired >= 5 && (t.top3 / t.fired) < 0.25);
  console.log(JSON.stringify(review, null, 2));
} catch(e) {
  console.error("Error:", e);
}

File: scripts/check.py
import json
import sys

try:
    with open('parsed.json', 'rb') as f:
        raw = f.read()
    
    # It might be utf-16le
    try:
        if raw.startswith(b'\xff\xfe'):
            text = raw.decode('utf-16le')
        else:
            text = raw.decode('utf-8')
    except Exception as e:
        print("Decode error:", e)
        text = raw.decode('utf-16le', errors='ignore')
    
    # try parsing
    # strip control characters that might break json except \n \r \t
    import re
    text = re.sub(r'[\x00-\x08\x0b\x0c\x0e-\x1f]', '', text)
    if text.startswith('\ufeff'):
        text = text[1:]
        
    data = json.loads(text)
    tags = data.get('tagStats', [])
    review = [t for t in tags if t.get('fired', 0) >= 5 and (t.get('top3', 0) / t.get('fired', 1)) < 0.25]
    print(json.dumps(review, ensure_ascii=False, indent=2))
except Exception as e:
    print(e)
